src/CommunityToolkit.Aspire.RavenDB.Client/RavenDBClientExtension.cs

```diff
@@ -0,0 +1,251 @@
+
+        var settings = new RavenDBSettings(connectionUrls, databaseName);
+        if (certificate != null)
+            settings.Certificate = certificate;
+
+        builder.AddRavenDBClient(settings, serviceKey: null);
+    }
+
+    /// <summary>
+    /// Registers <see cref="IDocumentStore"/> and the associated <see cref="IDocumentSession"/> and <see cref="IAsyncDocumentSession"/>
+    /// instances for connecting to an existing or new RavenDB database with RavenDB.Client, identified by a unique service key.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder"/> used to add services.</param>
+    /// <param name="serviceKey">A unique key that identifies this instance of the RavenDB client service.</param>
+    /// <param name="settings">The settings required to configure the <see cref="IDocumentStore"/>.</param>
+    /// <remarks>Note: If <see cref="RavenDBSettings.DatabaseName"/> is not specified and <see cref="RavenDBSettings.CreateDatabase"/>
+    /// is set to 'false', <see cref="IDocumentSession"/> and <see cref="IAsyncDocumentSession"/> will not be registered.</remarks>
+    public static void AddKeyedRavenDBClient(
+        this IHostApplicationBuilder builder,
+        object serviceKey,
+        RavenDBSettings settings)
+    {
+        ValidateSettings(builder, settings);
+
+        builder.AddRavenDBClient(settings, serviceKey);
+    }
+
+    /// <summary>
+    /// Registers <see cref="IDocumentStore"/> and the associated <see cref="IDocumentSession"/> and <see cref="IAsyncDocumentSession"/>
+    /// instances for connecting to an existing or new RavenDB database with RavenDB.Client, identified by a unique service key.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder"/> used to add services.</param>
+    /// <param name="serviceKey">A unique key that identifies this instance of the RavenDB client service.</param>
+    /// <param name="connectionUrls">The URLs of the RavenDB cluster nodes to connect to.</param>
+    /// <param name="databaseName">Optional: the name of an existing database to connect to.
+    /// If not specified, <see cref="IDocumentSession"/> and <see cref="IAsyncDocumentSession"/> will not be registered,
+    /// as a database context is required for session creation.</param>
+    /// <param name="certificate">Optional. The certificate for the server.</param>
+    /// <remarks>Note: If <see cref="RavenDBSettings.DatabaseName"/> is not specified and <see cref="RavenDBSettings.CreateDatabase"/>
+    /// is set to 'false', <see cref="IDocumentSession"/> and <see cref="IAsyncDocumentSession"/> will not be registered.</remarks>
+    public static void AddKeyedRavenDBClient(
+        this IHostApplicationBuilder builder,
+        object serviceKey,
+        string[] connectionUrls,
+        string? databaseName,
+        X509Certificate2? certificate = null)
+    {
+        ValidateSettings(builder, connectionUrls);
```

Comment:
can we have single `ValidateSettings` method, here we can create the `RavenDBSettings` first, and then pass them to ValidateSettings

Reply:
done